agent/structs/intention.go

```diff
@@ -336,9 +331,9 @@ func (x *Intention) String() string {
 
 // EstimateSize returns an estimate (in bytes) of the size of this structure when encoded.
 func (x *Intention) EstimateSize() int {
-	// 60 = 36 (uuid) + 16 (RaftIndex) + 4 (Precedence) + 4 (DefaultPort)
-	size := 60 + len(x.Description) + len(x.SourceNS) + len(x.SourceName) + len(x.DestinationNS) +
-		len(x.DestinationName) + len(x.SourceType) + len(x.Action) + len(x.DefaultAddr)
+	// 56 = 36 (uuid) + 16 (RaftIndex) + 4 (Precedence)
+	size := 56 + len(x.Description) + len(x.SourceNS) + len(x.SourceName) + len(x.DestinationNS) +
+		len(x.DestinationName) + len(x.SourceType) + len(x.Action)
```

Comment:
Does this need to account for the space of the deprecated fields, or will those not take up any space? Would adding `codec:",omitempty"` allow us to ignore the fields here?

I need to read up more about MessagePack to see if `codec:"-"` would be an option for these fields.

Reply:
I am not sure if it matters a whole lot. The various `EstimateSize` methods you may see around the codebase are usually used during replication to determine how many items to apply to the secondary dc in a single batch. These calculations only need to be very rough estimations and not accurate. If off by a few bytes it shouldn't matter.


https://github.com/hashicorp/go-msgpack/blob/ad60660ecf9c5a1eae0ca32182ed72bab5807961/codec/encode.go#L653-L655

Looks like `codec:"-"` will never encode the field. `codec:",omitempty"` should work to not emit the zero values.

Reply:
Thanks! I'll make these `codec:",omitempty"` for now